### ISSUE TITLE
Initializing Action causes crash on device

### ISSUE THREAD

**Issue by danzimm** [ReactiveCocoa/ReactiveCocoa#2733]
The following snippet crashes when running on an iPhone 6S running iOS 9.2:
```
        Action { (user: String) in
            return SignalProducer<String, NoError>(value: "")
        }
```
This was compiled with clang-703.0.21 using Xcode 7.3 beta 3 (7D141l)

An example project with the crash is over [here](http://files.danz.im/ActionCrash.zip).

The stack trace is:
```
  * frame #0: 0x00000001831997d0 libsystem_platform.dylib`_os_lock_corruption_abort
    frame #1: 0x00000001831997d0 libsystem_platform.dylib`_OSSpinLockLockSlow$VARIANT$wfe + 164
    frame #2: 0x000000010013ee0c ReactiveCocoa`specialized SignalType.combineLatestWith<A> (Signal<A1, A.Error>) -> Signal<(A.Value, A1), A.Error> [inlined] generic specialization <Swift.Optional<ReactiveCocoa.Bag<ReactiveCocoa.Observer<Swift.Bool, ReactiveCocoa.NoError>>>> of ReactiveCocoa.Atomic.(self=(spinLock = -1862220803, _value = 0xa90157f6a9bc5ff8)) () -> () + 1308 at Atomic.swift:33 [opt]
    frame #3: 0x000000010013ee04 ReactiveCocoa`specialized SignalType.combineLatestWith<A> (Signal<A1, A.Error>) -> Signal<(A.Value, A1), A.Error> [inlined] generic specialization <(Swift.Bool, Swift.Bool), ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa> of ReactiveCocoa.Signal.init ((ReactiveCocoa.Observer<A, B>) -> Swift.Optional<ReactiveCocoa.Disposable>) -> ReactiveCocoa.Signal<A, B> at Signal.swift:82 [opt]
    frame #4: 0x000000010013ee04 ReactiveCocoa`specialized SignalType.combineLatestWith<A> (Signal<A1, A.Error>) -> Signal<(A.Value, A1), A.Error> [inlined] generic specialization <(Swift.Bool, Swift.Bool), ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa> of ReactiveCocoa.Signal.__allocating_init ((ReactiveCocoa.Observer<A, B>) -> Swift.Optional<ReactiveCocoa.Disposable>) -> ReactiveCocoa.Signal<A, B> at Signal.swift:0 [opt]
    frame #5: 0x000000010013ee04 ReactiveCocoa`specialized SignalType.combineLatestWith<A> (otherSignal=<unavailable>, self=<unavailable>) -> Signal<(A.Value, A1), A.Error> + 1300 at Signal.swift:401 [opt]
    frame #6: 0x0000000100166468 ReactiveCocoa`_TPA__TTSg5GC13ReactiveCocoa6SignalSbOS_7NoError_GS0_SbS1__S_10SignalTypeS__S1_S1_s9ErrorTypeS__Sb_Sb___TFE13ReactiveCocoaPS_10SignalType17combineLatestWithurfGCS_6Signalqd__wx5Error_GS1_Twx5Valueqd___wxS2__ + 32 at Signal.swift:0 [opt]
    frame #7: 0x0000000100190084 ReactiveCocoa`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed, Arg[3] = Owned To Guaranteed, Arg[4] = Owned To Guaranteed, Arg[5] = Owned To Guaranteed> of generic specialization <ReactiveCocoa.SignalProducer<Swift.Bool, ReactiveCocoa.NoError> with ReactiveCocoa.SignalProducer<Swift.Bool, ReactiveCocoa.NoError> : ReactiveCocoa.SignalProducerType in ReactiveCocoa, ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa, Swift.Bool, Swift.Bool, ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa, (Swift.Bool, Swift.Bool), ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa> of (extension in ReactiveCocoa):ReactiveCocoa.SignalProducerType.((liftRight in _BB0E5F533EEE68215FAAD3E87E37BD99) <A, B, C, D where B1: Swift.ErrorType, D1: Swift.ErrorType> ((ReactiveCocoa.Signal<A.Value, A.Error>) -> (ReactiveCocoa.Signal<A1, B1>) -> ReactiveCocoa.Signal<C1, D1>) -> (ReactiveCocoa.SignalProducer<A1, B1>) -> ReactiveCocoa.SignalProducer<C1, D1>).(closure #1).(closure #1).(closure #1).(closure #1) + 424 at SignalProducer.swift:418
    frame #8: 0x00000001001907f4 ReactiveCocoa`function signature specialization <Arg[0] = Owned To Guaranteed, Arg[1] = Owned To Guaranteed, Arg[2] = Owned To Guaranteed, Arg[3] = Owned To Guaranteed, Arg[4] = Owned To Guaranteed, Arg[5] = Owned To Guaranteed> of generic specialization <ReactiveCocoa.SignalProducer<Swift.Bool, ReactiveCocoa.NoError> with ReactiveCocoa.SignalProducer<Swift.Bool, ReactiveCocoa.NoError> : ReactiveCocoa.SignalProducerType in ReactiveCocoa, ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa, Swift.Bool, Swift.Bool, ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa, (Swift.Bool, Swift.Bool), ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa> of (extension in ReactiveCocoa):ReactiveCocoa.SignalProducerType.((liftRight in _BB0E5F533EEE68215FAAD3E87E37BD99) <A, B, C, D where B1: Swift.ErrorType, D1: Swift.ErrorType> ((ReactiveCocoa.Signal<A.Value, A.Error>) -> (ReactiveCocoa.Signal<A1, B1>) -> ReactiveCocoa.Signal<C1, D1>) -> (ReactiveCocoa.SignalProducer<A1, B1>) -> ReactiveCocoa.SignalProducer<C1, D1>).(closure #1).(closure #1).(closure #1) + 1408 at SignalProducer.swift:245
    frame #9: 0x0000000100175778 ReactiveCocoa`generic specialization <ReactiveCocoa.SignalProducer<Swift.Bool, ReactiveCocoa.NoError> with ReactiveCocoa.SignalProducer<Swift.Bool, ReactiveCocoa.NoError> : ReactiveCocoa.SignalProducerType in ReactiveCocoa, ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa, Swift.Bool, Swift.Bool, ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa, (Swift.Bool, Swift.Bool), ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa> of (extension in ReactiveCocoa):ReactiveCocoa.SignalProducerType.((liftRight in _BB0E5F533EEE68215FAAD3E87E37BD99) <A, B, C, D where B1: Swift.ErrorType, D1: Swift.ErrorType> ((ReactiveCocoa.Signal<A.Value, A.Error>) -> (ReactiveCocoa.Signal<A1, B1>) -> ReactiveCocoa.Signal<C1, D1>) -> (ReactiveCocoa.SignalProducer<A1, B1>) -> ReactiveCocoa.SignalProducer<C1, D1>).(closure #1).(closure #1) + 1152 at SignalProducer.swift:245
    frame #10: 0x000000010018e0f0 ReactiveCocoa`specialized specialized SignalProducerType.(lift<A, B where ...> ((Signal<A.Value, A.Error>) -> Signal<A1, B1>) -> SignalProducer<A1, B1>).(closure #1) + 60 at SignalProducer.swift:261 [opt]
    frame #11: 0x000000010018e0b4 ReactiveCocoa`specialized specialized SignalProducerType.(lift<A, B where ...> ((Signal<A.Value, A.Error>) -> Signal<A1, B1>) -> SignalProducer<A1, B1>).(closure #1) [inlined] generic specialization <(Swift.Bool, Swift.Bool), ReactiveCocoa.NoError with ReactiveCocoa.NoError : Swift.ErrorType in ReactiveCocoa> of protocol witness for ReactiveCocoa.SignalProducerType.startWithSignal ((ReactiveCocoa.Signal<A.Value, A.Error>, ReactiveCocoa.Disposable) -> ()) -> () in conformance <A, B where B: Swift.ErrorType> ReactiveCocoa.SignalProducer<A, B> : ReactiveCocoa.SignalProducerType in ReactiveCocoa at SignalProducer.swift:232 [opt]
    frame #12: 0x000000010018e0b4 ReactiveCocoa`specialized specialized SignalProducerType.(outerDisposable=0x000000012ed5f580) -> Signal<A1, B1>) -> SignalProducer<A1, B1>).(closure #1) + 1324 at SignalProducer.swift:386 [opt]
    frame #13: 0x00000001001bda80 ReactiveCocoa`Action.init<A where ...> (enabledIf : A1, (A) -> SignalProducer<B, C>) -> Action<A, B, C> + 36 at SignalProducer.swift:261 [opt]
    frame #14: 0x00000001001bda5c ReactiveCocoa`Action.init<A where ...> (enabledIf=<unavailable>) -> SignalProducer<B, C>) -> Action<A, B, C> + 3296 at Action.swift:77 [opt]
    frame #15: 0x00000001001bdf88 ReactiveCocoa`Action.init((A) -> SignalProducer<B, C>) -> Action<A, B, C> + 912 at Action.swift:85 [opt]
    frame #16: 0x00000001000888b0 ActionCrash`AppDelegate.application(application=0x000000012ee09b70, launchOptions=nil, self=0x000000012ed06e60) -> Bool + 988 at AppDelegate.swift:27
    frame #17: 0x0000000100088be8 ActionCrash`@objc AppDelegate.application(UIApplication, didFinishLaunchingWithOptions : [NSObject : AnyObject]?) -> Bool + 216 at AppDelegate.swift:0
    frame #18: 0x000000018829d704 UIKit`-[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 400
    frame #19: 0x00000001884cc130 UIKit`-[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 2904
    frame #20: 0x00000001884d04b8 UIKit`-[UIApplication _runWithMainScene:transitionContext:completion:] + 1672
    frame #21: 0x00000001884cd5c0 UIKit`-[UIApplication workspaceDidEndTransaction:] + 168
    frame #22: 0x0000000184aeb790 FrontBoardServices`-[FBSSerialQueue _performNext] + 184
    frame #23: 0x0000000184aebb10 FrontBoardServices`-[FBSSerialQueue _performNextFromRunLoopSource] + 56
    frame #24: 0x00000001834ecefc CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
    frame #25: 0x00000001834ec990 CoreFoundation`__CFRunLoopDoSources0 + 540
    frame #26: 0x00000001834ea690 CoreFoundation`__CFRunLoopRun + 724
    frame #27: 0x0000000183419680 CoreFoundation`CFRunLoopRunSpecific + 384
    frame #28: 0x0000000188296580 UIKit`-[UIApplication _run] + 460
    frame #29: 0x0000000188290d90 UIKit`UIApplicationMain + 204
    frame #30: 0x0000000100089418 ActionCrash`main + 144 at AppDelegate.swift:13
    frame #31: 0x0000000182fba8b8 libdyld.dylib`start + 4
```

**Comment by eimantas** [comment]
It doesn't crash on Xcode 7.2.1. So it's probably just an Xcode/clang beta thing.

**Comment by larryonoff** [comment]
The same happens for me on Xcode 7.3, iPhone 6S iOS 9.3, swift 2.2.

**Comment by danzimm** [comment]
This is associated with #2751

**Comment by larryonoff** [comment]
@danzimm thanks! compiling RAC from swift-2.2 branch.

**Comment by larryonoff** [comment]
The issue fixed in swift-2.2 branch.
